lib/AST/Type.cpp

```diff
@@ -1499,6 +1499,13 @@ bool TypeBase::hasDynamicMemberLookupAttribute() {
       ctx.evaluator, HasDynamicMemberLookupAttributeRequest{canTy}, false);
 }
 
+bool TypeBase::hasDynamicCallableAttribute() {
+  auto canTy = getCanonicalType();
```

Comment:
Could you check for `mayHaveMembers` here and bail out if it returns `false`? This will help avoid filling up the evaluator's cache with lots of function types. The same check could also be added for `hasDynamicMemberLookupAttribute`.

Reply:
Sure, Fixed @hamishknight!

Reply:
Thanks!